### PR TITLE
maddy: init at 0.4.3

### PR DESCRIPTION
**Pull request by lxea**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/maddy/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{ pkgs, buildGoModule, fetchFromGitHub }:
```
Comment (SuperSandro2000) [inline pkgs/servers/maddy/default.nix:1]:
```suggestion
{ lib, buildGoModule, fetchFromGitHub }:
```

File: pkgs/servers/maddy/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      -X github.com/foxcpp/maddy.Version=${version}
+  '';
+
+  subPackages = [ "cmd/maddy" "cmd/maddyctl" ];
+
+  meta = {
+    description = "Composable all-in-one mail server";
+    homepage = "https://foxcpp.dev/maddy";
+    license = pkgs.lib.licenses.gpl3Plus;
+    maintainers = with pkgs.lib.maintainers; [ lxea ];
```
Comment (SuperSandro2000) [inline pkgs/servers/maddy/default.nix:27]:
```suggestion
    license = licenses.gpl3Plus;
    maintainers = with lib.maintainers; [ lxea ];
```

File: pkgs/servers/maddy/default.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  vendorSha256 = "16laf864789yiakvqs6dy3sgnnp2hcdbyzif492wcijqlir2swv7";
+
+  buildFlagsArray = ''
+    -ldflags=
+      -X github.com/foxcpp/maddy.Version=${version}
+  '';
+
+  subPackages = [ "cmd/maddy" "cmd/maddyctl" ];
+
+  meta = {
```
Comment (SuperSandro2000) [inline pkgs/servers/maddy/default.nix:23]:
```suggestion
  meta = with lib; {
```

File: pkgs/servers/maddy/default.nix [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    repo = "maddy";
+    rev = "v${version}";
+    sha256 = "1mi607hl4c9y9xxv5lywh9fvpybprlrgqa7617km9rssbgk4x1v7";
+  };
+
+  vendorSha256 = "16laf864789yiakvqs6dy3sgnnp2hcdbyzif492wcijqlir2swv7";
+
+  buildFlagsArray = ''
+    -ldflags=
+      -X github.com/foxcpp/maddy.Version=${version}
```
Comment (SuperSandro2000) [inline pkgs/servers/maddy/default.nix:18]:
```suggestion
      -s
      -w
      -X github.com/foxcpp/maddy.Version=${version}
```

File: pkgs/servers/maddy/default.nix [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  src = fetchFromGitHub {
+    owner = "foxcpp";
+    repo = "maddy";
+    rev = "v${version}";
+    sha256 = "1mi607hl4c9y9xxv5lywh9fvpybprlrgqa7617km9rssbgk4x1v7";
+  };
+
+  vendorSha256 = "16laf864789yiakvqs6dy3sgnnp2hcdbyzif492wcijqlir2swv7";
+
+  buildFlagsArray = ''
```
Comment (SuperSandro2000) [inline pkgs/servers/maddy/default.nix:16]: This should be an array not a string.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 112005` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>maddy</li>
</ul>
</details>

File: pkgs/servers/maddy/default.nix [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  src = fetchFromGitHub {
+    owner = "foxcpp";
+    repo = "maddy";
+    rev = "v${version}";
+    sha256 = "1mi607hl4c9y9xxv5lywh9fvpybprlrgqa7617km9rssbgk4x1v7";
+  };
+
+  vendorSha256 = "16laf864789yiakvqs6dy3sgnnp2hcdbyzif492wcijqlir2swv7";
+
+  buildFlagsArray = [ ''
```
Comment (sikmir) [inline pkgs/servers/maddy/default.nix:16]: IMHO, less lines is more clear in such case:
```suggestion
  buildFlagsArray = [ "-ldflags=-s -w -X github.com/foxcpp/maddy.Version=${version}" ];
```

**Pull request merged**